src/lib/util/Config.ts

```diff
@@ -123,21 +123,29 @@ export class Config {
 			return true;
 		}
 
-		return !(await this.getConfigValue('disabled', engine, this.typeChecker.booleanCheck));
+		return !(await this.getBooleanConfigValue('disabled', engine));
 	}
 
-	public async getSupportedLanguages(engine: ENGINE): Promise<string[]> {
-		return await this.getConfigValue('supportedLanguages', engine, this.typeChecker.stringArrayCheck);
+	public getSupportedLanguages(engine: ENGINE): Promise<any> {
+		return this.getStringArrayConfigValue('supportedLanguages', engine);
 	}
 
-	public async getTargetPatterns(engine: ENGINE): Promise<string[]> {
-		return await this.getConfigValue('targetPatterns', engine, this.typeChecker.stringArrayCheck);
+	public getTargetPatterns(engine: ENGINE): Promise<any> {
+		return this.getStringArrayConfigValue('targetPatterns', engine);
 	}
 
-	private async getConfigValue(propertyName: string, engine: ENGINE, typeChecker: (any, string, ENGINE) => boolean): Promise<string[]> {
+	protected getBooleanConfigValue(propertyName: string, engine: ENGINE): Promise<boolean> {
+		return this.getConfigValue(propertyName, engine, this.typeChecker.booleanCheck);
+	}
+
+	protected getStringArrayConfigValue(propertyName: string, engine: ENGINE): Promise<string[]> {
+		return this.getConfigValue(propertyName, engine, this.typeChecker.stringArrayCheck);
+	}
+
+	private async getConfigValue(propertyName: string, engine: ENGINE, typeChecker: (any, string, ENGINE) => boolean): Promise<any> {
 		let ecc = this.getEngineConfig(engine);
 		// If the config specifies property, use those.
-		if (ecc && ecc[propertyName] && typeChecker(ecc[propertyName], propertyName, engine)) {
+		if (ecc && (ecc[propertyName] !== undefined) && typeChecker(ecc[propertyName], propertyName, engine)) {
```

Comment:
Bug fix.

Reply:
In JS, it's typical to do `!= null` instead of `!== undefined` or `!== null`. That's because `undefined` and `null` are `==` equivalent but not `===` equivalent, and you typically want to treat `undefined` and `null` the same way. This feels like it's one of those times, because I imagine we'd want to treat a value of `null` in the Config as equivalent to the property being entirely absent.

Reply:
It seems like that difference could be left to the typechecker. Some values might want to allow nulls and others not, but changing it to != null would preclude allowing that granularity.

Reply:
Honestly, the typechecker itself feels like something that could be refactored away through using the tools TS provides, most notably Type Guarding (https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards). We should probably target this file for a refactor the next time we have an opportunity.

But that's neither here nor there. You've convinced me that it makes sense to treat `undefined` and `null` as distinct values, but there should be a comment explaining why we're doing this unconventional thing.